apps/worker/src/app/workflow/usecases/execute-bridge-job/execute-bridge-job.usecase.ts

```diff
@@ -288,6 +288,17 @@ export class ExecuteBridgeJob {
             lastSeenDate: message.lastSeenDate || null,
             lastReadDate: message.lastReadDate || null,
           };
+        } else {
+          /*
+           * Provide fallback state for in-app messages to satisfy framework inAppResultSchema validation
+           * when message is not found (e.g., cancelled jobs, nv-5120)
+           */
+          output = {
+            seen: false,
+            read: false,
+            lastSeenDate: null,
+            lastReadDate: null,
+          };
```

Comment:
this is the solution for the current issue.

Reply:
I think this place needs refactoring as a follow-up task. If a step is not executed, the framework should handle it, not the worker. So the logic needs to be moved there with step result defaults.

Reply:
https://linear.app/novu/issue/NV-5165/framework-refactor-unexecuted-step-logic